crates/turbo-trace/src/tracer.rs

```diff
@@ -211,15 +211,36 @@ impl Tracer {
                     debug!("built in: {:?}", err);
                 }
                 Err(err) => {
-                    // Try to resolve the import as a type import via `@/types/<import>`
-                    let type_package = format!("@types/{}", import);
-                    let resolved_type_import = resolver
-                        .resolve(file_path, type_package.as_str())
+                    if !import.starts_with(".") {
+                        // Try to resolve the import as a type import via `@/types/<import>`
+                        let type_package = format!("@types/{}", import);
+                        debug!("trying to resolve type import: {}", type_package);
+                        let resolved_type_import = resolver
+                            .resolve(file_dir, type_package.as_str())
+                            .ok()
+                            .and_then(|resolved| resolved.into_path_buf().try_into().ok());
+
+                        if let Some(resolved_type_import) = resolved_type_import {
+                            debug!("resolved type import succeeded");
+                            files.push(resolved_type_import);
+                            continue;
+                        }
+                    }
+
+                    // Also try without the extension just in case the wrong extension is used
+                    let without_extension = Utf8Path::new(import).with_extension("");
```

Comment:
Just calling out that this will convert something like `types.d.ts` to `types.d` [source](https://docs.rs/camino/latest/camino/struct.Utf8PathBuf.html#method.extension), is that the desired behavior?

Reply:
So I originally had an explicit check for `types.d.ts` that stripped it all the way back to `types` but when I ran this on the other repo, that didn't fix any errors. I suspect oxc-resolver also treats `types.d` as the name, so this should work just fine